sdk/identity/identity/review/identity.api.md

```diff
@@ -32,10 +32,16 @@ export const AuthenticationErrorName = "AuthenticationError";
 
 // @public
 export class AuthorizationCodeCredential implements TokenCredential {
-    constructor(tenantId: string, clientId: string, clientSecret: string | undefined, authorizationCode: string, redirectUri: string, options?: IdentityClientOptions);
+    constructor(clientId: string, clientSecret: string, authorizationCode: string, redirectUri: string, options?: AuthorizationCodeCredentialOptions);
+    constructor(clientId: string, authorizationCode: string, redirectUri: string, options?: AuthorizationCodeCredentialOptions);
```

Comment:
I don't think that tenantId should be optional here but should be added into the overloads. It is only optional in the case that the specified clientId is a multi-tenanted application (which most will not be). Adding the overloads gives us the ability to document this.

Reply:
Just to check - you're saying that we need both constructors? 

So these:
```typescript
constructor(clientId: string, clientSecret: string, authorizationCode: string, redirectUri: string, options?: AuthorizationCodeCredentialOptions);
constructor(clientId: string, authorizationCode: string, redirectUri: string, options?: AuthorizationCodeCredentialOptions);

```

AND

```typescript
constructor(tenantId: string, clientId: string, clientSecret: string, authorizationCode: string, redirectUri: string, options?: AuthorizationCodeCredentialOptions);
constructor(tenantId: string, clientId: string, authorizationCode: string, redirectUri: string, options?: AuthorizationCodeCredentialOptions);
```

Or are you saying that 'tenantId' is itself an optional parameter in all constructor overloads?

Reply:
The four you have above look correct. The top 2 would be documented as only working for multi-tenant apps.

Reply:
Turns out the original suggestion makes things ambiguous so we're going with making tenantId a required parameter (ie, no overload) and then just documenting that there's a default value ('common') that can be used.

Reply:
Done - tenantId is now a required parameter with a type of ' string |"common"".

I also added in this doc comment:
'common' may be used when dealing with multi-tenant scenarios.